packages/authgear-core/src/container.ts

```diff
@@ -69,6 +72,16 @@ export abstract class BaseContainer<T extends BaseAPIClient> {
    */
   delegate?: ContainerDelegate;
 
+  /**
+   * @public
+   */
+  sessionState: SessionState;
+
+  /**
+   * @public
+   */
+  onSessionStateChangedListeners: OnSessionStateChangedListener[];
```

Comment:
This should be internal. Instead, expose `_addOnSessionStateChangedListener` and `_removeOnSessionStateChangedListener` for public consumption.